### ISSUE TITLE
[Java] A false negative about the rule FormatString

### ISSUE THREAD

**Issue by Zustin** [google/error-prone#2486]
Hi, I found a false negative about the rule [FormatString](https://errorprone.info/bugpattern/FormatString). The minimal case is below. Error prone should report a warning at line 4 because variable *this.fs* is an invalid format string. Thanks for your consdieration.

```java
final String fs = "%r";
public void func() {
   String str;
   str = String.format(this.fs, "hello"); // should report a warning here
}
```

Expected outcome: A warning
Actual outcome: No warning

**Comment by Zustin** [comment]
@cpovirk, Hi Chris, could you please help check this issue? Thanks for your kind help and consideration.

**Comment by cushon** [comment]
The check only validates format strings if the value of the string is available at compile-time, i.e. if it's a [compile-time constant](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.28).
